search/tasks_test.py

```diff
@@ -40,3 +54,121 @@ def test_increment_document_integer_field_task(mocked_api):
     increment_document_integer_field(*indexing_api_args)
     assert mocked_api.increment_document_integer_field.call_count == 1
     assert mocked_api.increment_document_integer_field.call_args[0] == indexing_api_args
+
+
+def test_wrap_retry_exception():
+    """wrap_retry_exception should raise RetryException when other exceptions are raised"""
+    with assert_not_raises():
+        with wrap_retry_exception(KeyError):
+            # Should not raise an exception
+            pass
+
+
+@pytest.mark.parametrize("matching", [True, False])
+def test_wrap_retry_exception_matching(matching):
+    """A matching exception should raise a RetryException"""
+
+    class SubError(KeyError):
+        """Use a subclass to assert isinstance use"""
+
+    def raise_thing():
+        """raise the exception"""
+        if matching:
+            raise SubError()
+        else:
+            raise TabError()
+
+    matching_exception = RetryException if matching else TabError
+    with pytest.raises(matching_exception):
+        with wrap_retry_exception(KeyError):
+            raise_thing()
+
+
+def test_index_post_with_comments(mocker):
+    """index_post should call the api function of the same name"""
+    index_post_mock = mocker.patch('search.indexing_api.index_post_with_comments')
+    wrap_mock = mocker.patch('search.tasks.wrap_retry_exception')
+    post_id = 'post_id'
+    index_post_with_comments.delay(post_id)
+
+    index_post_mock.assert_called_once_with(post_id)
+    wrap_mock.assert_called_once_with(PrawcoreException, PRAWException)
+
+
+def test_index_channel(mocker, settings, user):
+    """index_channel should index all posts of a channel"""
+    settings.INDEXING_API_USERNAME = user.username
+    index_post_mock = mocker.patch('search.tasks.index_post_with_comments', autospec=True)
+    api_mock = mocker.patch('channels.api.Api', autospec=True)
+
+    expected_exception = ZeroDivisionError
```

Comment:
I read your earlier comment. In this case I think it's best to pull this patching logic out into a separate fixture to make the actual tests clearer. The fixture can patch `celery.app.task.Task.replace`, `search.tasks.group`,`search.tasks.chain`, and any other celery library functions that we might need to patch in the future. The return value for that fixture can also include this exception so the `pytest.raises` assertion would look something like this: `with pytest.raises(celery_mock.expected_exception):`. Anything we can do to isolate this finicky library patching and clarify the actual intent of the tests would be really helpful here IMO.

Reply:
Sounds good